packages/flame/test/extensions/vector2_test.dart

```diff
@@ -208,4 +208,42 @@ void main() {
       expectDouble(position.screenAngle(), math.pi / 2);
     });
   });
+
+  group('projection', () {
+    test('Project onto longer vector', () {
+      final u = Vector2(5, 2);
+      final v = Vector2(10, 0);
+      final result = u.projection(v);
+      expect(result, Vector2(5, 0));
+    });
+
+    test('Project onto shorter vector', () {
+      final u = Vector2(5, 2);
+      final v = Vector2(2, 0);
+      final result = u.projection(v);
+      expect(result, Vector2(5, 0));
+    });
+
+    test('Project onto vector in other direction', () {
+      final u = Vector2(5, 2);
+      final v = Vector2(-10, 0);
+      final result = u.projection(v);
+      expect(result, Vector2(5, 0));
+    });
+  });
```

Comment:
Seems to be no tests for the `out` argument

Reply:
Fixed